src/Owin.Security.Providers.Salesforce/SalesforceAuthenticationHandler.cs

```diff
@@ -187,7 +187,12 @@ protected override Task ApplyResponseChallengeAsync()
             var state = Options.StateDataFormat.Protect(properties);
 
             var authorizationEndpoint =
-                $"{Options.Endpoints.AuthorizationEndpoint}?response_type={"code"}&client_id={Options.ClientId}&redirect_uri={HttpUtility.UrlEncode(redirectUri)}&display={"page"}&immediate={false}&state={Uri.EscapeDataString(state)}&scope={""}";
+                $"{Options.Endpoints.AuthorizationEndpoint}?response_type={"code"}&client_id={Options.ClientId}&redirect_uri={HttpUtility.UrlEncode(redirectUri)}&display={"page"}&immediate={false}&state={Uri.EscapeDataString(state)}";
+
+            if (Options.Scope.Count > 0)
```

Comment:
`Options` cannot be null as all the proceeding lines would fail.
Technically `Options.Scope` cannot be null as it is set to an empty list in the constructor and has a private setter.
However, I agree that the extra safety would be nice, how about this?

`Options.Scope?.Count > 0 ?? false`